components/brave_shields/browser/ad_block_component_service_manager.cc

```diff
@@ -251,6 +262,35 @@ void AdBlockComponentServiceManager::EnableFilterList(const std::string& uuid,
   UpdateFilterListPrefs(uuid, enabled);
 }
 
+void AdBlockComponentServiceManager::CheckAdBlockComponentsUpdate() {
+  auto runner = base::SequencedTaskRunner::GetCurrentDefault();
+
+  runner->PostDelayedTask(FROM_HERE, base::BindOnce([]() {
```

Comment:
As @atuchin-m noted at https://github.com/brave/brave-core/pull/20557/files#r1365810455,
It would be great if we could update in a bunch.

Reply:
agreed... I don't think we have any API exposed for this though.

The best thing I've found so far is this [`UpdateClient::Update`](https://source.chromium.org/chromium/chromium/src/+/main:components/update_client/update_client.h;l=507;drc=4a6bf24b15fdb49a018a12af2025321691f87e1a;bpv=0;bpt=1) method that takes multiple component IDs, but we can't call it directly (would have to plumb multiple levels of patching). There's also [`CrxUpdateService::CheckForUpdates`](https://source.chromium.org/chromium/chromium/src/+/main:components/component_updater/component_updater_service.cc;l=396;drc=4a6bf24b15fdb49a018a12af2025321691f87e1a;bpv=0;bpt=1), although we don't have the ability to customize the list of components checked. Maybe that's ok?

 @atuchin-m do you have any recommendations?

Reply:
1.  My point is we'd better optimize the number of requests to the backend to check the update. If we want to update the set of components, ideally we should done the one request. 
2. `UpdateClient::Update` looks like an internal thing. 
3. `CheckForUpdates` looks better, maybe it's okay to check the update of all the brave components.
4.  Anyway, scheduling a number of task with random delays doesn't looks reasonable to me. 10 sec usually is not enough to process an update. Do you sure that the next PostTasks don't cancel the previous ones?

Reply:
Also `BraveOnDemandUpdater` uses [priority](https://github.com/chromium/chromium/blob/master/components/component_updater/component_updater_service.h#L155-L158) `FOREGROUND`, that means there is no chance to combine the requests together. `CheckForUpdates` uses the same priority internally.

I believe the first thing to do is to understand the right priority to use. if it's `BACKGROUND` then it's look okay to use `CheckForUpdates`. Otherwise I believe we need to add `CrxUpdateService::OnDemandUpdateMultipleIds` via chromium_src [here](https://github.com/brave/brave-core/blob/master/chromium_src/components/component_updater/component_updater_service.h)

@iefremov do we have someone who is familiar with component_updater internals?

Reply:
I'm somewhat familiar with component_updater internals.